guides/common/modules/proc_upgrading-the-external-database.adoc

```diff
@@ -0,0 +1,24 @@
+[id="Upgrading_the_External_Database_{context}"]
+= Upgrading the External Database
+
+You can upgrade an external database from {EL} 7 to {EL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+* Create a new {EL} 8 based host for PostgreSQL server that follows the external database on {EL} 8 documentation.
+For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/using-databases#migrating-to-a-rhel-8-version-of-postgresql_using-postgresql[Migrating to a RHEL 8 version of PostgreSQL].
+
+.Procedure
+. On a virtual machine, take a snapshot.
+. On a physical machine, create a backup.
+. If you keep the same hostname on the new PostgreSQL server, update the DNS record for the host to prevent the need to do updates on {Project} itself.
+. If the new PostgreSQL server has a new host name or a different CNAME, update the reference to the external database on {Project}:
```

Comment:
I think this is confusing. Let's first start at the requirement. The requirement is that the Foreman server can reach the DB server.

There are multiple ways to achieve this.

First of all, a new hostname is used and the procedure below uses the installer to change the configs accordingly.

Then the other is that the same hostname is used. To make sure it reaches the right server, you can use two techniques.

You described changing the DNS record, but this has some potential issues. Any DNS record has a TTL (Time To Live), which allows any DNS server to cache it. If you don't plan for it and you have a high TTL value (86400 seconds, or 1 day, is common) then it can take a long time for the migration to propagate. A common technique is to plan for it, lower the TTL in advance (at least $TTL) and then change it. Once it's stable it can be raised again.

However during the migration it may still resolve to the old host if you have multiple caching servers. That's why many admins actually prefer to move the IP to the new server instead. So you use some temporary IP to completely provision & prepare the server and then add the old IP while removing it from the old server (or shutting that down).

Some also combine that and do migrate to a new IP but add both IPs to the same server during the migration so the stale DNS cache isn't an issue.

In my mind those are all fairly standard operating procedures so I wonder how much of it should be written out.

Reply:
@evgeni related to my comment above regarding this point. Your thoughts?

Reply:
I think I'd slightly reword this, making it less ambiguous:
* create a backup 
* restore the backup on the new server
* If {Project} can reach the new database server via the old name, no further changes are required. Otherwise reconfigure {Project} to use the new name: <installer invocation>

All further details are left as an exercise to the reader, as we do not know how their setup looks like.

Reply:
> create a backup

Should we recommend an offline backup? So stop all services and then create a backup to prevent data loss?

Reply:
Given https://github.com/theforeman/foreman-documentation/blob/a9d7313f01d8fbe0d7a4a88db72583a24d1f7385/guides/common/assembly_using-external-databases.adoc#L10-L12
I would tend to say "nah, not our pint"

(but we probably should add a similar sentence to non-Satellite builds of the documentation)